apps/web/src/app/(dashboard)/admin/users/[id]/disable-user-dialog.tsx

```diff
@@ -0,0 +1,150 @@
+'use client';
+
+import { useState } from 'react';
+
+import { useRouter } from 'next/navigation';
+
+import { Trans, msg } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+
+import type { User } from '@documenso/prisma/client';
+import { TRPCClientError } from '@documenso/trpc/client';
+import { trpc } from '@documenso/trpc/react';
+import { Alert, AlertDescription, AlertTitle } from '@documenso/ui/primitives/alert';
+import { Button } from '@documenso/ui/primitives/button';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@documenso/ui/primitives/dialog';
+import { Input } from '@documenso/ui/primitives/input';
+import { useToast } from '@documenso/ui/primitives/use-toast';
+
+export type DisableUserDialogProps = {
+  className?: string;
+  user: User;
+};
+
+export const DisableUserDialog = ({ className, user }: DisableUserDialogProps) => {
+  const { toast } = useToast();
+  const { _ } = useLingui();
+
+  const router = useRouter();
+
+  const [email, setEmail] = useState('');
+
+  const { mutateAsync: disableUser, isLoading: isDisablingUser } =
+    trpc.admin.disableUser.useMutation();
+
+  const onDisableAccount = async () => {
+    try {
+      await disableUser({
+        id: user.id,
+        email,
+      });
+
+      toast({
+        title: _(msg`Account disabled`),
+        description: _(msg`The account has been disabled successfully.`),
+        duration: 5000,
+      });
+
+      router.push('/admin/users');
```

Comment:
I don't think a push is necessary since we're not deleting a resource here

Reply:
Yeah, I tried some stuff here (close the modal) and forgot to update this. 😅